### PR TITLE
Add functions to userdata example

### PR DESCRIPTION
**Pull request by ghost**
# Add functions to userdata example

### CODE REVIEW

**Comment by ghost** [comment]
# Here is an example of what I contribute


```C

int book_sayhello(lua_State *L) {
  printf("\r\nHello world from book_sayhello()!\r\n");

  return 0;
}

```

```Lua

print(b.title)
print(b.author)
print(b.pages)
print(b.year)

b:sayhello()
b:sayhello()

```

**Comment by Miqueas** [comment]
Ok, I can finally review this, gonna check it out

File: 06-userdata.c [hunk 1]
```diff
@@ -5,72 +5,76 @@
...
+
+int book_sayhello(lua_State *L) {
+  printf("\r\nHello world from book_sayhello()!\r\n");
+
+  return 0;
+}
 
 int luaopen_userdata(lua_State *L) {
-  lua_pushcfunction(L, book_new);
+  lua_register(L, LIBBOOK_NAME, book__new);
```
Comment (Miqueas) [inline 06-userdata.c:34]: `lua_register()` creates global "items" in the Lua side, I usually don't do that and isn't something that I wanna "teach"

File: 06-userdata.c [hunk 2]
```diff
@@ -5,72 +5,76 @@
...
-  // stack: Book(userdata)
-  lua_setmetatable(L, 1);
 
-  return 1;
-}
+  /*Add table to add variables and functions*/  
+  luaL_newmetatable(L, LIBBOOK_NAME);
+
+  /*Add index*/
+  lua_pushvalue(L, -1); lua_setfield(L, -2, "__index");
```
Comment (Miqueas) [inline 06-userdata.c:59]: This is _tecnically_ the same to what I did, but whatever

File: 06-userdata.c [hunk 3]
```diff
@@ -5,72 +5,76 @@
...
+  /*
+  lua api functions commonly work with the top of the stack as arguments as well as output
+  */
+  lua_pushstring(L, b->title); lua_setfield(L, -2, "title");
+  lua_pushstring(L, b->author); lua_setfield(L, -2, "author");
+  lua_pushinteger(L, b->pages); lua_setfield(L, -2, "pages");
+  lua_pushinteger(L, b->year); lua_setfield(L, -2, "year");
+
+  /*Add metatable to userdata*/
+  lua_setmetatable(L, 1);
```
Comment (Miqueas) [inline 06-userdata.c:77]: Well... The _"funny"_ thing of using a function is... Uh, idk

**Review by Miqueas — COMMENTED** [review]
Well, to be honest, I'm not merging this since the only one relevant change I see is the function/method of the `Book` userdata and I can implement that without problems :+1:

**Pull request closed without merging**